libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -1634,28 +1634,33 @@ static void ProcessInterconnect(vtr::string_internment* strings, pugi::xml_node
     num_complete = count_children(Parent, "complete", loc_data, ReqOpt::OPTIONAL);
     num_direct = count_children(Parent, "direct", loc_data, ReqOpt::OPTIONAL);
     num_mux = count_children(Parent, "mux", loc_data, ReqOpt::OPTIONAL);
-    num_interconnect = num_complete + num_direct + num_mux;
+    num_partial = count_children(Parent, "partial", loc_data, ReqOpt::OPTIONAL);
+    num_interconnect = num_complete + num_direct + num_mux + num_partial;
 
     mode->num_interconnect = num_interconnect;
     mode->interconnect = new t_interconnect[num_interconnect];
 
     i = 0;
-    for (L_index = 0; L_index < 3; L_index++) {
+    for (L_index = 0; L_index < 4; L_index++) {
         if (L_index == 0) {
             Cur = get_first_child(Parent, "complete", loc_data, ReqOpt::OPTIONAL);
         } else if (L_index == 1) {
             Cur = get_first_child(Parent, "direct", loc_data, ReqOpt::OPTIONAL);
-        } else {
+        } else if (L_index == 2) {
```

Comment:
Is this magic number related to enum e_interconnect?  That enum has values 1,2,3,4, but this code is checking for values 0,1,2,3, so they are different, but they look related.  Should there be an enum for the arch file reading here?

Reply:
Okay, I see what's going on now.  It's just looping through the number of items in the enum, but the actual L_index variable isn't really used.

To clean up the code a bit, perhaps we should add something like this:
```
std::map<enum e_interconnect, std::string> e_interconnect_to_arch_name = {{COMPLETE_INTERC, "complete"}, {DIRECT_INTERC, "direct"} ,...};
```

Then you can change this loop to:
```
for (auto e : e_interconnect_to_arch_name) {
}
```